yellowbrick/text/postag.py

```diff
@@ -20,125 +20,371 @@
-            'LS'   : 'yellow',
-            'MD'   : 'yellow',
-            'PDT'  : 'yellow',
-            'RP'   : 'yellow',
-            'SYM'  : 'yellow',
-            'TO'   : 'yellow',
-            'None' : 'off'
+    
+    def _uni_tag_map(self):
+        """
+        Returns a Universal Dependencies part-of-speech tag map.
+        """
+        return {
+            "noun" : 0,
+            "verb" : 0,
+            "adjective" : 0,
+            "adverb" : 0,
+            "adposition" : 0,
+            "determiner" : 0,
+            "pronoun" : 0,
+            "conjunction" : 0,
+            "infinitive" : 0,
+            "punctuation" : 0,
+            "number" : 0,
+            "interjection" : 0,
+            "symbol": 0,
+            "other" : 0
         }
 
-    def colorize(self, token, color):
+    def _handle_universal(self, X):
         """
-        Colorize text
+        Scan through the corpus to compute counts of each Universal
+        Dependencies part-of-speech.
 
         Parameters
         ----------
-        token : str
-            A str representation of
-
+        X : list or generator
+            Should be provided as a list of documents or a generator
+            that yields a list of documents that contain a list of 
+            sentences that contain (token, tag) tuples.
         """
-        return self.COLORS[color].format(token)
-
-    def transform(self, tagged_tuples):
+        for tagged_doc in X:
+            for tagged_sent in tagged_doc:
+                for _, tag in tagged_sent:
+                    # combine proper and regular nouns
+                    if tag in ["NOUN", "PROPN"]:
+                        self.tag_map["noun"] += 1
+                    elif tag == "ADJ":
+                        self.tag_map["adjective"] += 1
+                    elif tag == "VERB":
+                        self.tag_map["verb"] += 1 
+                    # include particles with adverbs
+                    elif tag in ["ADV", "PART"]:
+                        self.tag_map["adverb"] += 1
+                    elif tag == "ADP":
+                        self.tag_map["adposition"] += 1
+                    elif tag == "PRON":
+                        self.tag_map["pronoun"] += 1
+                    elif tag == "CCONJ":
+                        self.tag_map["conjunction"] += 1
+                    elif tag == "PUNCT":
+                        self.tag_map["punctuation"] += 1
+                    elif tag == "DET":
+                        self.tag_map["determiner"] += 1
+                    elif tag == "NUM":
+                        self.tag_map["number"] += 1
+                    elif tag == "INTJ":
+                        self.tag_map["interjection"] += 1
+                    elif tag == "SYM":
+                        self.tag_map["symbol"] += 1
+                    elif tag == "SPACE":
+                        continue
+                    else:
+                        self.tag_map["other"] += 1                
```

Comment:
These handle methods are extensive and well done. I do very much like how you're distilling the much larger tagsets into a manageable number of larger, human-understandable word classes for use in the visualizer (and how you described that distillation in the documentation). 

An alternative to the case statements is to use a jump table as follows:

```python
jump = {
    # combine proper and regular nouns 
    "NOUN": "noun", "PROPN": "noun",
    "ADJ": "adjective",
    "VERB": "verb",
    # include particles with adverbs 
    "ADV": "adverb", "PART": "adverb",
    "ADP": "adposition",
    "PRON": "pronoun",
    "CCONJ": "conjunction",
    "PUNCT": "punctuation",
    "DET": "determiner",
    "NUM": "number",
    "INTJ": "interjection",
    "SYM": "symbol",
}

for tagged_doc in X:
    for tagged_sent in tagged_doc:
        for _, tag in tagged_sent:
            if tag == "SPACE":
                continue
            self.tag_map_[jump.get(tag, "other")] += 1 
```

This change means that in the worst case (everything is "other") will have the same performance as the best case (everything is a "noun"). This change may be noticible for much larger corpora (e.g. gigabytes of data). If you like this alternate formulation (e.g. maybe more readable/maintainable), then that's a good reason to change; but otherwise we can always mark this is technical debt in the case that this visualizer is used on much larger datasets.

Reply:
Changed to jump table to avoid incurring technical debt.